### PR TITLE
maven clean up, fix checkstyle reporting

### PR DESCRIPTION
**Pull request by SlowMo24**
messing with Maven Dependency mess. upgrading some dependencies, removing unused reporting (maven site), enabling personal checkstyle sheet

### CODE REVIEW

**Comment by SlowMo24** [comment]
@rtroilo might be consulted first because maven is always tricky and hi knows

**Comment by SlowMo24** [comment]
also I am sorry this commit comes AFTER the codestyle week. But it was only now requested by @tyrasd  :-)

File: pom.xml [hunk 1]
```diff
@@ -71,6 +79,54 @@
...
+            <version>${pmd.version}</version>
+          </dependency>
+          <dependency>
+            <groupId>net.sourceforge.pmd</groupId>
+            <artifactId>pmd-java</artifactId>
+            <version>${pmd.version}</version>
+          </dependency>
+          <dependency>
+            <groupId>net.sourceforge.pmd</groupId>
+            <artifactId>pmd-javascript</artifactId>
```
Comment (tyrasd) [inline pom.xml]: do we need the javascript module as well?

File: Jenkinsfile [hunk 2]
```diff
@@ -132,7 +126,7 @@ pipeline {
           
           //warnings plugin
           rtMaven.run pom: 'pom.xml', goals: '--batch-mode -V -e checkstyle:checkstyle pmd:pmd pmd:cpd findbugs:findbugs com.github.spotbugs:spotbugs-maven-plugin:3.1.7:spotbugs -Dmaven.repo.local=.m2'
-
+
```
Comment (tyrasd) [inline Jenkinsfile]: normally, we try to avoid unnecessary white space like here…

File: pom.xml (hunk 1 above)
Comment (SlowMo24) [reply pom.xml]: ok, your right, should have checked before copy paste... sorry

File: Jenkinsfile (hunk 2 above)
Comment (SlowMo24) [reply Jenkinsfile]: can't explain where they come from. thanks for noticing. Will catch up on this...

**Pull request merged**